vspace/src/main/java/edu/asu/diging/vspace/web/staff/SpaceOverviewController.java

```diff
@@ -0,0 +1,172 @@
+                List<IModuleLinkDisplay> tempSpaceToModulelinkList = new ArrayList<>();
+                tempSpaceToModulelinkList.add(link);
+                AllSpaceToModulelinksListMap.put(link.getLink().getSpace().getId(), tempSpaceToModulelinkList);
+            } else {
+                List<IModuleLinkDisplay> tempSpaceToModulelinkList = AllSpaceToModulelinksListMap
+                        .get(link.getLink().getSpace().getId());
+                tempSpaceToModulelinkList.add(link);
+                AllSpaceToModulelinksListMap.put(link.getLink().getSpace().getId(), tempSpaceToModulelinkList);
+            }
+        });
+
+        List<ISpaceLinkDisplay> AllSpaceToSpacelinksList = spaceLinkDisplayRepository
+                .findAllSpaceLinkDisplaysForSpace();
+        Map<String, List<ISpaceLinkDisplay>> AllSpaceToSpacelinksListMap = new HashMap<>();
+
+        AllSpaceToSpacelinksList.forEach(link -> {
+            if (!AllSpaceToSpacelinksListMap.containsKey(link.getLink().getSourceSpace().getId())) {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = new ArrayList<>();
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceToSpacelinksListMap.put(link.getLink().getSourceSpace().getId(), tempSpaceToSpacelinkList);
+            } else {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = AllSpaceToSpacelinksListMap
+                        .get(link.getLink().getSourceSpace().getId());
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceToSpacelinksListMap.put(link.getLink().getSourceSpace().getId(), tempSpaceToSpacelinkList);
+            }
+        });
+
+        List<SpaceLinkDisplay> AllSpaceLinkForGivenOrNullSpaceStatusList = spaceLinkDisplayRepository
+                .findAllSpaceLinksForGivenOrNullSpaceStatus(SpaceStatus.PUBLISHED);
+        Map<String, List<ISpaceLinkDisplay>> AllSpaceLinkForGivenOrNullSpaceStatustMap = new HashMap<>();
+
+        AllSpaceLinkForGivenOrNullSpaceStatusList.forEach(link -> {
+            if (!AllSpaceLinkForGivenOrNullSpaceStatustMap.containsKey(link.getLink().getSourceSpace().getId())) {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = new ArrayList<>();
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceLinkForGivenOrNullSpaceStatustMap.put(link.getLink().getSourceSpace().getId(),
+                        tempSpaceToSpacelinkList);
+            } else {
+                List<ISpaceLinkDisplay> tempSpaceToSpacelinkList = AllSpaceLinkForGivenOrNullSpaceStatustMap
+                        .get(link.getLink().getSourceSpace().getId());
+                tempSpaceToSpacelinkList.add(link);
+                AllSpaceLinkForGivenOrNullSpaceStatustMap.put(link.getLink().getSourceSpace().getId(),
+                        tempSpaceToSpacelinkList);
+            }
+        });
+        if (spaceList != null && !spaceList.isEmpty()) {
+            spaceList.forEach(s -> {
```

Comment:
since this is a huge for loop, make it a proper for loop and give `s` a better name; this is really confusing.